consensus/propagation/reactor.go

```diff
@@ -63,12 +69,33 @@ func NewReactor(self p2p.ID, tracer trace.Tracer, store *store.BlockStore, mempo
 		ProposalCache: NewProposalCache(store),
 		mempool:       mempool,
 		started:       atomic.Bool{},
+		ctx:           ctx,
+		cancel:        cancel,
 	}
 	reactor.BaseReactor = *p2p.NewBaseReactor("BlockProp", reactor, p2p.WithIncomingQueueSize(ReactorIncomingMessageQueueSize))
 
 	for _, option := range options {
 		option(reactor)
 	}
+
+	// start the catchup routine
+	go func() {
+		// TODO dynamically set the ticker depending on how many blocks are missing
+		ticker := time.NewTicker(time.Second)
```

Comment:
we should set this to roughly the block time

1 second is too frequent and we'll end up requesting too many parts, which starts a negative feedback loop

Reply:
what's the target block time? 3s? 6s?